pkg/types/clusterconfig/clusterconfig.go

```diff
@@ -428,44 +427,29 @@ func (cc *Config) Validate(awsClient *aws.Client) error {
 	}
 
 	if cc.Spot != nil && *cc.Spot {
-		cc.AutoFillSpot(awsClient)
-		chosenInstance := aws.InstanceMetadatas[*cc.Region][*cc.InstanceType]
-		compatibleSpots := CompatibleSpotInstances(awsClient, chosenInstance, cc.SpotConfig.MaxPrice, _spotInstanceDistributionLength)
-		if len(compatibleSpots) == 0 {
-			return errors.Wrap(ErrorNoCompatibleSpotInstanceFound(chosenInstance.Type), InstanceTypeKey)
-		}
+		cc.FillEmptyFields(awsClient)
 
-		compatibleInstanceCount := 0
+		chosenInstance := aws.InstanceMetadatas[*cc.Region][*cc.InstanceType]
```

Comment:
nit: maybe this should be called `primaryInstance` or `primaryInstanceType`, what do you think?

Reply:
Agreed.

Reply:
I changed `chosenInstance` to `primaryInstance` and `*cc.InstanceType` to `primaryInstanceType`.